src/Navigation/NavLayout.js

```diff
@@ -0,0 +1,24 @@
+import Media from "react-media";
+import NavbarMobile from "./NavbarMobile";
+import NavbarDesktop from "./NavbarDesktop";
+const NavLayout = () => {
+  return (
+    <>
+      <Media
+        queries={{
+          small: "(max-width: 768px)",
+          large: "(min-width: 769px)"
+        }}
+      >
+        {(matches) => (
+          <>
+            {matches.small && <NavbarMobile />}
+            {matches.large && <NavbarDesktop />}
+          </>
+        )}
+      </Media>
+    </>
+  );
+};
+

```

Comment:
I found this way of writing media queries quite interesting!.......might wanna discusss this approach!

Reply:
i found this library when i was reading about react-router. the same team made this, the main advantage i feel is that if your desktop and mobile versions are quite different then it is hard to manage through just css.

also this loads the component accordingly, and the other version is not even rendered, so its not just a 'display: none'

although there is some code repetition in this approach, so im still not sure what's the best approach